### PR TITLE
adding fallback when indexer is down to fetch nfts owners

### LINKED ISSUE #659: Implement back up query & notification for stale NFT indexer data
**Describe the bug**
If the indexer data is stale, nft ownership/quantity may not be accurate. We should implement an indexer health check and backup queries where possible, to fetch current data. 

**To Reproduce**
1. Check if indexer data is stale by querying the /v1/health endpoint
2. If indexer data is stale (or hardcode stale health check result) , perform an NFT transaction
3. See nft ownership fail to update and ability to send again

**Expected behavior**
User should be notified on collection inventory page (see Teloscan for stale data notification) if data is stale and a backup query should be performed to update individual NFT data on the details page

**Tasks**

- [x] Notification of stale data (NFT inventory page)
- [x] Use ethers to query 721 contract for `ownerOf` to update 
- [x] Use ethers to query 1155 contract for `balanceOf` to update quantity

**Additional context**
- there is already a health check in code and backup for balances

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #659

## Description
This Pull Request addresses the issue of stale NFT indexer data. It implements a backup query mechanism and user notifications for cases where the indexer data is stale. The main enhancements include:
- Adding an indexer health check.
- Implementing backup queries for fetching the current NFT owner.
- Notifying users on the collection inventory page about the indexer being not healthy.
- Querying ERC721 and ERC1155 contracts directly for `ownerOf` and `balanceOf` updates when needed.
These changes ensure that NFT ownership and quantity are accurately reflected, even when the indexer data is not up-to-date.

## Test Scenarios
Currently, the indexer is up to date so I created a way to simulate the indexer is not healthy.
- go to [this page](https://deploy-preview-671--wallet-staging.netlify.app/) and login
- go to demo section (left-bottom) 
- go to "Simulation indexer down"
- press the button "Simulate indexer down" and wait for it to go down
- press Homepage link (don't refresh the page)
- login again
- go to inventory page
  - you should get a warning notification about indexer being down
- go to any ERC1155 detail page
- check the owner list
- transfer the NFT and wait for the page to reload NFT owners
  - it only updates your balance of that ERC1155
  - other owner balances stay the same (because we don't use the indexer)

## Videocapture

https://github.com/telosnetwork/telos-wallet/assets/4420760/5a742f1e-8daa-49fa-9a66-a68208814396

## Checklist:
-   [x] Implemented indexer health check and user notifications for stale data.
-   [x] Added backup queries to fetch current NFT data from contracts directly.
-   [x] Ensured that the solution works for both ERC721 and ERC1155 contracts.
-   [x] Conducted thorough testing to confirm the accuracy of NFT data.
-   [x] Reviewed and cleaned up the code in the affected areas.
-   [x] No new warnings generated by the changes.
-   [x] All changes merged and published in downstream modules.
-   [x] Code and comments reviewed for clarity and comprehensibility.
-   [x] English text included in the translation file or new issue created for required translations.
-   [x] Mobile functionality and responsiveness tested.
-   [x] Appropriate test coverage added.

### CODE REVIEW

**Review by donnyquixotic — DISMISSED** [review]

File: src/antelope/chains/EVMChainSettings.ts [hunk 1]
```diff
@@ -383,6 +411,14 @@ export default abstract class EVMChainSettings implements ChainSettings {
 
         this.processNftContractsCalldata(response.contracts);
         const shapedNftData = this.shapeNftRawData(shapedIndexerNftData, response.contracts);
+
+        // if the owner does not comes from the indexer, let's set it
+        shapedNftData.forEach((nft) => {
```
Comment (ezra-sg) [inline src/antelope/chains/EVMChainSettings.ts:416]: this should only be set for erc721, which is why line 409 has a ternary like 
`owner: isErc1155 ? undefined : nftResponse.owner,`
so here we should also only set owner if it's 1155

File: test/jest/utils/antelope/store-account.ts [hunk 2]
```diff
@@ -12,7 +12,11 @@ const loggedNativeAccount = jest.fn().mockImplementation(() => null);
 const currentAccount = jest.fn().mockImplementation(() => null);
 const currentEvmAccount = jest.fn().mockImplementation(() => null);
 const currentIsLogged = jest.fn().mockImplementation(() => true);
-const getAccount = jest.fn().mockImplementation(() => null);
+const getAccount = jest.fn().mockImplementation(() => ({
+    authenticator: jest.fn().mockImplementation(() => ({
+        web3Provider: jest.fn().mockImplementation(() => 'do you need to get deeper?'),
```
Comment (ezra-sg) [inline test/jest/utils/antelope/store-account.ts:17]: 😆

**Comment by ezra-sg** [comment]
when i click Demos, it kicks me back to the home page

**Review by donnyquixotic — DISMISSED** [review]

File: src/antelope/chains/EVMChainSettings.ts (hunk 1 above)
Comment (donnyquixotic) [reply src/antelope/chains/EVMChainSettings.ts]: I updated to use `owner: nftResponse.owner ?? account`

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**